plugin/core/sessions.py

```diff
@@ -802,6 +803,18 @@ def run_command(self, command: Mapping[str, Any]) -> Promise:
             )
         )
 
+    def _apply_workspace_edit_async(self, edit: Any) -> Promise:
+        """
+        Apply workspace edits, and return a promise that resolves on the async thread again after the edits have been
+        applied.
+        """
+        changes = parse_workspace_edit(edit)
+        return Promise.from_main_thread().then(
+            lambda _: apply_workspace_edit(self.window, changes).then(
+                lambda _: Promise.from_async_thread()
+            )
+        )
```

Comment:
This doesn't look very pretty or readable. Maybe would make sense to use line-continuation marks here? 
```suggestion
        return Promise.from_main_thread() \
            .then(lambda _: apply_workspace_edit(self.window, changes)) \
            .then(lambda _: Promise.from_async_thread())
```